File: perftest/soa.c
#include <stdio.h>



struct Voxel { 
		float Red;
		float Green;
		float Blue; 
		double lightray[11];
}; 


int main() { 
	int i,j;
	struct Voxel scene1[200][200]; 
	struct Voxel scene2[200][200]; 

   for (i = 0; i < 200; i++) {
                for (j = 0; j < 200; j++) {

                        scene1[i][j].Red = 1.3*(i+j);
                        scene1[i][j].Green = 1.4*(i+j);
                        scene1[i][j].Blue = 1.5*(i+j);
 
                        scene2[i][j].Red = scene2[i][j].Green = scene2[i][j].Blue = 1;
                }
        }

	for (i = 0; i < 199; i++) {
    		for (j = 0; j < 199; j++) {
      			scene1[i][j].Red = (scene1[i][j].Red * scene2[i+1][j].Red)/2; 
      			scene1[i][j].Green = (scene1[i][j].Green * scene2[i+1][j].Green)/2; 
      			scene1[i][j].Blue = (scene1[i][j].Blue * scene2[i+1][j].Blue)/2; 
		}
  	}
	return 0; 
}

File: perftest/soa.py
#! /usr/bin/python
import numpy as np

class Voxel:
  Red = 0.0
  Green = 0.0
  Blue = 0.0
  lightray = None
 


#scene1 = np.empty([200,200]); 
#scene2 = np.empty([200,200]); 


#scene1 = [[0]*200 for j in range[200]]
scene1 = {}
#scene2 = [[0 for i in range(200)] for j in range[200]]
scene2 = {}

for i in range(0,200):
    for j in range(0,200):
        v1 = Voxel()
        v2 = Voxel()
        v1.Red = 1.3*(i+j)
        v1.Green = 1.4*(i+j)
        v1.Blue = 1.5*(i+j)
        v2.Red = 1
        v2.Green =  1
        v2.Blue = 1
        scene1[i,j] = v1
        scene2[i,j] = v2

for i in range(0, 199):
    for j in range(0, 199):
        v1 = scene1[i,j]
        v2 = scene2[i+1,j]
        v1.Red = (v1.Red * v2.Red)/2 
        v1.Green = (v1.Green * v2.Green)/2
        v2.Blue = (v1.Blue * v2.Blue)/2

